src/GovUk.Education.ExploreEducationStatistics.Common/Utils/CompressionUtils.cs

```diff
@@ -13,6 +13,34 @@ namespace GovUk.Education.ExploreEducationStatistics.Common.Utils;
 
 public static class CompressionUtils
 {
+    public static Stream GetCompressionStream(
+        Stream targetStream,
+        string contentEncoding,
+        CompressionMode compressionMode)
+    {
+        switch (contentEncoding)
+        {
+            case ContentEncodings.Gzip:
+                {
+                    return new GZipStream(targetStream, compressionMode, leaveOpen: true);
+                }
+            case ContentEncodings.Zstd:
+                {
+                    switch (compressionMode)
+                    {
+                        case CompressionMode.Compress:
+                            return new CompressionStream(targetStream, level: 11, leaveOpen: true);
+                        case CompressionMode.Decompress:
+                            return new DecompressionStream(targetStream, leaveOpen: true);
+                        default:
+                            throw new ArgumentException($"Unsupported compression mode {compressionMode}");
+                    }
+                }
+            default:
+                throw new NotSupportedException($"Content encoding {contentEncoding} is not supported");
+        }
+    }
```

Comment:
Could we simplify things below now by reusing this in `CompressToStream` and `DecompressToStream`?

Reply:
Hey @benoutram - I think that ideally we would deduplicate this all and tidy up.  Probably starting with removing all CompressToStream / DecompressToStream usages that still use intermediate Streams and then removing the methods altogether.  Then when everything's using GetCompressionStream, look at the common patterns and suck them right back into CompressionUtils again!  Probably calling them CompressToStream and DecompressToStream!!